parser.go

```diff
@@ -0,0 +1,206 @@
+package godotenv
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"strings"
+	"unicode"
+)
+
+const (
+	charComment       = '#'
+	prefixSingleQuote = '\''
+	prefixDoubleQuote = '"'
+
+	exportPrefix = "export"
+)
+
+func parseBytes(src []byte, out map[string]string) error {
+	cutset := src
```

Comment:
After testing and playing around with this, I believe the following will pass all tests and provide the expected behavior. We should not be replacing "\r" with "\n" but "\r\n" is safe to replace with "\n" when parsing the file, and it does fix the problematic behavior you were encountering. @joho 

```suggestion
	src = bytes.Replace(src, []byte("\r\n"), []byte("\n"), -1)
	cutset := src
```